pygmt/tests/test_clib_put_vector.py

```diff
@@ -90,6 +90,62 @@ def test_put_vector_mixed_dtypes():
                 npt.assert_allclose(newy, y)
 
 
+def test_put_vector_string_dtype():
+    """
+    Passing string type data to a dataset.
+    """
+    vectors = np.array(
+        [
+            ["10", "20.0", "-30.0", "3.5e1"],  # numbers
+            ["10W", "30.50E", "30:30W", "40:30:30.500E"],  # longitudes
+            ["10N", "30.50S", "30:30N", "40:30:30.500S"],  # latitudes
+            # datetimes
+            ["2021-02-03", "2021-02-03T04", "2021-02-03T04:05:06.700", "T04:50:06.700"],
+        ]
+    )
+
+    expected_vectors = [
+        [10.0, 20.0, -30.0, 35],  # numbers
+        [-10, 30.5, -30.5, 40.508472],  # longitudes
+        [10, -30.50, 30.5, -40.508472],  # latitudes
+        # datetimes
+        ["2021-02-03", "2021-02-03T04", "2021-02-03T04:05:06.700", "T04:50:06.700"],
+    ]
+
+    expected_dtypes = [np.double, np.double, np.double, np.str_]
+
+    for i, j in itertools.combinations_with_replacement(range(3), r=2):
+        # TODO: Change range(3) to range(4)
```

Comment:
The idea of the test is to pass different combinations of strings to GMT, let GMT write it to a temporary file, and check if the output is the same as expected (i.e., if `vector=expected_vetcor`.

Still has trouble testing datetime strings, when the text file contains strings,
```
dtype=[("x", np.str_), ("y", np.str_)],
```
doesn't works as I expect.

Reply:
Fixed in bf7587cb2ca1b34ade13f0239fc078ca4279c332, but could still be improved.